src/session.ts

```diff
@@ -302,38 +304,37 @@ function rememberLogin(): Promise<Session> {
   })
 }
 
-function refresh(): Promise<void> {
-  return fetchJSON<Session>('/account/info', { cache: 'reload' })
-  .then((data: Session) => {
+async function refresh(): Promise<void> {
+  try {
+    const data = await fetchJSON<Session>('/account/info', { cache: 'reload' })
     session = data
     storeSession(data)
+    announce.set(data.announce)
```

Comment:
You are thinking of an `announce` data here in case people would miss the websocket message? I don't know if it's necessary, I guess we should see that with @ornicar if needed.

If you want to start implementing the websocket handler here you can btw.

Reply:
Yeah - on the desktop site, from what I understand, the `announce` data is injected on page load or sent once via websocket. My concern is that mobile users who are not online when the announcement is made will miss the websocket message, so they'll never know about it otherwise.